### PR TITLE
refactor(Achievement): migrate achievement-creator helpers to user_id reads

### PR DESCRIPTION
**Pull request by wescopeland**
Another PR with no functional changes. Migrates all _achievement-creator.php_ `Achievements.Author` reads to rely on `user_id`.

### CODE REVIEW

File: app/Helpers/database/achievement-creator.php [hunk 1]
```diff
@@ -1,70 +1,87 @@
...
+
+    return $userAuthoredAchievements
+        ->groupBy('game.system.Name')
+        ->map(function ($achievements, $systemName) {
+            return [
+                'ConsoleName' => $systemName,
+                'AchievementCount' => $achievements->count(),
+            ];
+        })
+        ->sortByDesc('AchievementCount')
```
Comment (Jamiras) [inline app/Helpers/database/achievement-creator.php:29]: The left side has a secondary sort on ConsoleName

Comment (wescopeland) [reply app/Helpers/database/achievement-creator.php]: Thanks, I missed that. Resolved in latest.

File: app/Helpers/database/achievement-creator.php [hunk 2]
```diff
@@ -1,70 +1,87 @@
...
-        'author' => $username,
-        'achievementFlag' => AchievementFlag::OfficialCore,
-    ])->toArray();
+    $userAuthoredAchievements = Achievement::whereHas("game.system", function ($query) {
+        $query->whereNotIn("ID", [System::Hubs, System::Events]);
+      })
+        ->with("game.system")
+        ->where("user_id", $user->id)
+        ->where("Flags", AchievementFlag::OfficialCore)
+        ->get();
```
[the remark this answers is not in the document]
Comment (wescopeland) [reply app/Helpers/database/achievement-creator.php:19]: Great feedback. I was afraid to get too prescriptive with systems in the model files themselves, so I went with:

```php
$userAuthoredAchievements = $user->authoredAchievements()
    ->published()
    ->whereHas('game.system', function ($query) {
        $query->whereNotIn('ID', [System::Hubs, System::Events]);
    })
    ->with('game.system')
    ->get();
```

File: app/Helpers/database/achievement-creator.php [hunk 3]
```diff
@@ -1,70 +1,87 @@
...
-        'joinUsername' => $username,
-        'achievementFlag' => AchievementFlag::OfficialCore,
-    ])->toArray();
+    $userAuthoredAchievements = Achievement::whereHas("game.system", function ($query) {
+        $query->whereNotIn("ID", [System::Hubs, System::Events]);
+      })
+        ->with("game.system")
+        ->where("user_id", $user->id)
+        ->where("Flags", AchievementFlag::OfficialCore)
+        ->get();
```
[the remark this answers is not in the document]
Comment (wescopeland) [reply app/Helpers/database/achievement-creator.php:68]: Thanks, updated in latest.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**